### ISSUE TITLE
API tests issues

### ISSUE THREAD

**Issue by voropa** [NikitaDunyashenko/Qase_Diploma#10]
Вот такой результат при запуске всех тестов из класса DefectApiTests
<img width="1352" alt="image" src="https://user-images.githubusercontent.com/14195681/224765850-90cacd46-3e04-40ff-bf01-195f1de721e4.png">

**Comment by NikitaDunyashenko** [comment]
Да, тут только если запускать по сьютам. Там есть метод, который удаляет дефект и по алфавиту он идет первым, а создает дефект метод beforeClass, а не beforeMethod. Просто, если делать его именно beforeMethod, то тесты не совсем независимыми получатся. Мне тогда нужно будет в каждом тесте менять defect id и подбирать под очередность тестов, так как проект создается beforeClass, а дефект beforeMethod.

**Comment by voropa** [comment]
Это неправильно, что тесты запускаются только  по  сьютам внутри класса. Это по  сути  и делает их зависымими, поотому что нельзя запустить их вместе. Найдите решение, сделайте их  независимыми, чтобы они запускались  все вместе и  работали. Например, для этих тестов на удаление/редактирование дефетка, сделайте прекоддишн, который будет создавать дефект, брать его ID из респоонса и сохранять в переменную, а потом в тесте вы будете апдейтить / удалять дефект с таким ID.
а сейчас у вас захардкодан id = 1
<img width="1403" alt="image" src="https://user-images.githubusercontent.com/14195681/224781951-0d455770-e3d2-4764-a950-d4fa95411c3a.png">

**Comment by NikitaDunyashenko** [comment]
Олег, я не до конца пониманию, как можно взять из body id. Единственное, что мне приходит в голову - это extract.body.asString, запихнуть ее в стринговую переменную и потом парсить строку и превращать ее в только в id, но потом как передать этот id в другой метод, если она видна не будет, я не понимаю как. Можно в принципе сделать этот метод не void, а Integer, чтобы возвращал число, но это уже похоже не на precondition, а на метод, который мне нужно вызывать в самом тесте, чтобы получить это число. Короче, как-то сложно у меня все придумывается. Я бы просто создавал перед каждым тестом новый проект и дефект. Тогда все выполнялось бы и с id не надо было ничего придумывать

**Comment by voropa** [comment]
> Олег, я не до конца пониманию, как можно взять из body id. Единственное, что мне приходит в голову - это extract.body.asString, запихнуть ее в стринговую переменную и потом парсить строку и превращать ее в только в id, но потом как передать этот id в другой метод, если она видна не будет, я не понимаю как. Можно в принципе сделать этот метод не void, а Integer, чтобы возвращал число, но это уже похоже не на precondition, а на метод, который мне нужно вызывать в самом тесте, чтобы получить это число. Короче, как-то сложно у меня все придумывается. Я бы просто создавал перед каждым тестом новый проект и дефект. Тогда все выполнялось бы и с id не надо было ничего придумывать

https://stackoverflow.com/a/21279938
